flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/FlinkTypeFactory.scala

```diff
@@ -79,14 +83,27 @@ class FlinkTypeFactory(typeSystem: RelDataTypeSystem) extends JavaTypeFactoryImp
   }
 
   private def createAdvancedType(typeInfo: TypeInformation[_]): RelDataType = typeInfo match {
-    // TODO add specific RelDataTypes
-    // for PrimitiveArrayTypeInfo, ObjectArrayTypeInfo, CompositeType
+    case ct: CompositeType[_] =>
+      new CompositeRelDataType(ct, this)
+
+    // TODO add specific RelDataTypes for PrimitiveArrayTypeInfo, ObjectArrayTypeInfo
     case ti: TypeInformation[_] =>
       new GenericRelDataType(typeInfo, getTypeSystem.asInstanceOf[FlinkTypeSystem])
 
     case ti@_ =>
       throw TableException(s"Unsupported type information: $ti")
   }
+
+  override def createTypeWithNullability(
+      relDataType: RelDataType,
+      nullable: Boolean)
+    : RelDataType = relDataType match {
+    case composite: CompositeRelDataType =>
+      // at the moment we do not care about nullability
```

Comment:
`CompositeRelDataType` is the first custom type and thus not supported by the superclass (`JavaTypeFactory`) that's why I had to override this method. The super method does some copying logic and sets the nullability of every field of the record. At the moment every field in the Table API can be null anyway, so I skipped this logic here until we provide full nullabilty support through the entire API.